registry/auth/token/token.go

```diff
@@ -172,26 +172,27 @@ func (t *Token) VerifySigningKey(verifyOpts VerifyOptions) (signingKey crypto.Pu
 	// verifying the first one in the list only at the moment.
 	header := t.JWT.Headers[0]
 
-	signingKey, err = verifyCertChain(header, verifyOpts.Roots)
-	// NOTE(milosgajdos): if the x5c header is missing
-	// the token may have been signed by a JWKS.
-	if err != nil && err != jose.ErrMissingX5cHeader {
-		return
-	}
-
-	switch {
-	case header.JSONWebKey != nil:
-		signingKey, err = verifyJWK(header, verifyOpts)
-	case len(header.KeyID) > 0:
-		signingKey = verifyOpts.TrustedKeys[header.KeyID]
-		if signingKey == nil {
-			err = fmt.Errorf("token signed by untrusted key with ID: %q", header.KeyID)
+	signingKey, err := verifyCertChain(header, verifyOpts.Roots)
+	if err != nil {
+		// NOTE(milosgajdos): if the x5c header is missing
+		// the token may have been signed by a JWKS.
+		if err == jose.ErrMissingX5cHeader {
```

Comment:
we should probably use `errors.Is` here

Reply:
I was considering that, too. Yeah, can change that.